Add MetaShift dataset (#3900)
huggingface/datasets: datasets/metashift/README.md

---
annotations_creators:
- crowdsourced
language_creators:
- crowdsourced
languages:
- en
licenses:
- cc-by-4.0
multilinguality:
- monolingual
pretty_name: MetaShift
size_categories:
- unknown

> source_datasets: []

Reviewer: ```suggestion
source_datasets:
- original
```

> task_categories:
> - other
> task_ids:
> - other-other-domain-generalization;subpopulation-shifts-settings;assess-training-conflicts

Reviewer: ```suggestion
task_categories:
- image-classification
- other
task_ids:
- multi-label-image-classification
- other-other-domain-generalization
```

- **Repository:** [MetaShift repository](https://github.com/Weixin-Liang/MetaShift)
- **Paper:** [MetaShift paper](https://arxiv.org/abs/2202.06523v1)
- **Point of Contact:**  [Weixin Liang](mailto:wxliang@stanford.edu)

### Dataset Summary

The MetaShift dataset is a collection of 12,868 sets of natural images across 410 classes. It was created for understanding the performance of a machine learning model across diverse data distributions.

The authors leverage the natural heterogeneity of Visual Genome and  its annotations used to construct MetaShift.
The key idea is to cluster images using its metadata which provides context for each image.
For example : cats with cars or cats in bathroom.
The main advantage is the dataset contains many more coherent sets of data compared to other benchmarks.

Two important benefits of MetaShift :
- Contains orders of magnitude more natural data shifts than previously available.
- Provides explicit explanations of what is unique about each of its data sets and a distance score that measures the amount of distribution shift between any two of its data sets.

### Dataset Usage

The dataset has the following configuration parameters:

> - selected_classes: `list[string]`, list of the classes needed to generate the MetaShift dataset.

Reviewer 2: ```suggestion
- selected_classes: `list[string]`, optional, list of the classes to generate the MetaShift dataset for. If `None`, the list is equal to `['cat', 'dog', 'bus', 'truck', 'elephant', 'horse']`.
```

- attributes: `list[string]`, list of attributes classes needed to generate the Attributes dataset. You can find the full attribute ontology in the above link.
- with_image_metadata: `bool`, default=False, to include image metadata, set as True. This will give additional metadata about each image. See [Scene Graph](https://cs.stanford.edu/people/dorarad/gqa/download.html) for more information.
- image_subset_size_threshold: `int`, default=25, The number of images required to consider a subset. If the number of images are less than image_subset_size_threshold, then the subset is not considered.
- min_local_groups: `int`, default=5, The minimum number of local groups required to consider an object class.

Consider the following examples to get an idea of how you can use the configuration parameters :

1. To generate the MetaShift Dataset :
```python
load_dataset("metashift", selected_classes=['cat', 'dog', 'bus'])
```
The full object vocabulary and its hierarchy can be seen [here](https://github.com/Weixin-Liang/MetaShift/blob/main/dataset/meta_data/class_hierarchy.json).

> The default classes are _CLASSES = [ 'cat', 'dog', 'bus', 'truck', 'elephant', 'horse',]

Reviewer 2: ```suggestion
The default classes are `['cat', 'dog', 'bus', 'truck', 'elephant', 'horse']`
```

2. To generate the MetaShift-Attributes Dataset (subsets defined by subject attributes) :
```python
load_dataset("metashift", attributes_dataset = True, attributes=["dog(smiling)", "cat(resting)"])
```

> The default attributes are _ATTRIBUTES = ["cat(orange)", "cat(white)","dog(sitting)","dog(jumping)"]

Reviewer 2: ```suggestion
The default attributes are `["cat(orange)", "cat(white)", "dog(sitting)", "dog(jumping)"]`
```
